### PR TITLE
Tokenizer handle collections

### PR DESCRIPTION
**Pull request by kiyomih**
- update tokenizer so we can handle collections
- update EBNF and test files so to match
- update type checking to handle collections

### CODE REVIEW

**Review by scveloso — APPROVED** [review]
Looks good to me!
Edit: Just need to fix the conflicts

**Comment by kiyomih** [comment]
updated so conflicts are resolved & double checked tests pass!

File: src/ast/ReturnDecl.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default class ReturnDecl extends AstNode {
     }
 
     public typeCheck(): void {
-        const wasDeclared: boolean = this.typeTable.isValidType(this.returnType);
+        const wasDeclared: boolean = this.typeTable.isValidType(this.returnType.replace(/[\[\]]/g, ""));
```
Comment (jyoo980) [inline src/ast/ReturnDecl.ts:26]: I'm assuming this line of code just "unwraps" the collection? i.e. get `number` from `number[]` by killing the brackets?

**Review by jyoo980 — APPROVED** [review]
LGTM!

File: src/ast/ReturnDecl.ts (hunk 1 above)
Comment (kiyomih) [reply src/ast/ReturnDecl.ts:26]: yep- that's right!

**Pull request merged**